src/lib/y2ntp_client/widgets/main_widgets.rb

```diff
@@ -0,0 +1,273 @@
+require "yast"
+
+require "cwm/widget"
+require "cwm/table"
+
+require "y2ntp_client/dialog/pool"
+
+Yast.import "Confirm"
+Yast.import "NtpClient"
+Yast.import "Popup"
+
+module Y2NtpClient
+  module Widgets
+    # Widget for netconfig policy
+    class PolicyCombo < CWM::ComboBox
+      def initialize
+        textdomain "ntp-client"
+      end
+
+      def label
+        # TRANSLATORS: label for widget that allows to define if ntp configiration is only
+        # from its source or dynamically extended e.g. via DHCP
+        _("Configuration Source")
+      end
+
+      def help
+        # TODO: not written previously, but really deserve something
+        _("<b>Configuration Source</b> defines if network itself can adds ntp sources.")
+      end
+
+      def opt
+        [:editable]
+      end
+
+      def items
+        items = [
+          # combo box item
+          ["", _("Static")],
+          # combo box item
+          ["auto", _("Dynamic")]
+        ]
+        current_policy = Yast::NtpClient.ntp_policy
+        if !["", "auto"].include?(current_policy)
+          items << [current_policy, current_policy]
+        end
+
+        items
+      end
+
+      def init
+        self.value = Yast::NtpClient.ntp_policy
+      end
+
+      def validate
+        if value =~ /["']/
+          Yast::Popup.Error(_("Configuration source contain invalid character"))
```

Comment:
The user will hate you for not being told *which* character is invalid :D

```rb
# TRANSLATORS: single quote (') and double quote (") are invalid
_("Configuration Source may not contain single or double quotes")
```

BTW what about SetFocus to the offending widget?